### ISSUE TITLE
Unable to reproduce mAP scores on object detection

### ISSUE THREAD

**Issue by vadthyavath** [aws/sagemaker-python-sdk#2621]
### Discussed in https://github.com/aws/sagemaker-python-sdk/discussions/2620

<div type='discussions-op-text'>

<sup>Originally posted by **vadthyavath** September  6, 2021</sup>
After tuning, Sagemaker inbuilt object detection algorithm(SSD) reports mAP of 71% on the validation set. Then I deployed this model and made predictions on the test set. But I couldn't get mAP scores close to validation set.

So in order to validate mAP metric computation, I made predictions on the validation set and then computed mAP scores. Validation mAP score which I computed doesn't match with sagemaker reported mAP score.

Then to validate mAP scoring criteria, I have used Gluoncv code from here  (https://cv.gluon.ai/_modules/gluoncv/utils/metrics/voc_detection.html) and other repo from here https://github.com/microsoft/vision-evaluation/blob/e82f087a4993b8852b8873a2d6c8ec474faa95dd/vision_evaluation/evaluators.py#L373

But I was unable to replicate sagemaker mAP scores.
Could you please provide references to the code which sagemaker uses to compute mAP score so that we can reproduce the same on test set?

update: coco based mAP score also doesn't match.</div>

**Comment by mohamedzeidan2021** [comment]
Closing — the built-in algorithm this issue targets is on a retired framework.

Apache MXNet reached end of life in September 2023, and SageMaker Python SDK v3 has dropped MXNet framework support. The built-in **Object Detection - MXNet** algorithm container is unlikely to receive fixes for the reproducibility gap you reported.

For new object-detection work, AWS recommends the **Object Detection - TensorFlow** built-in algorithm or modern detection models via **SageMaker JumpStart** (YOLO, Faster R-CNN, DETR). If you can reproduce a similar mAP mismatch against one of those, please open a new issue naming the specific algorithm/model. Thanks for the detailed repro.
